packages/eslint-plugin-internal/tests/rules/no-dynamic-tests.test.ts

```diff
@@ -0,0 +1,207 @@
+  },
+});
+
+ruleTester.run('no-dynamic-tests', rule, {
+  invalid: [
+    // Function calls in test arrays
+    {
+      code: `
+ruleTester.run('test', rule, {
+  valid: [generateTestCases()],
+  invalid: [],
+});
+      `,
+      errors: [
+        {
+          column: 11,
+          line: 3,
+          messageId: 'noDynamicTests',
+        },
+      ],
+    },
+    {
+      code: `
+ruleTester.run('test', rule, {
+  valid: [],
+  invalid: [...getInvalidCases()],
+});
+      `,
+      errors: [
+        {
+          column: 13,
+          line: 4,
+          messageId: 'noDynamicTests',
+        },
+      ],
+    },
+    // Spread operator in test arrays
+    {
+      code: `
+ruleTester.run('test', rule, {
+  valid: [...validTestCases],
+  invalid: [],
+});
+      `,
+      errors: [
+        {
+          column: 11,
+          line: 3,
+          messageId: 'noDynamicTests',
+        },
+      ],
+    },
+    {
+      code: `
+ruleTester.run('test', rule, {
+  valid: [...validTestCases.map(t => t.code)],
+  invalid: [],
+});
+      `,
+      errors: [
+        {
+          column: 11,
+          line: 3,
+          messageId: 'noDynamicTests',
+        },
+      ],
+    },
+    // Simple identifiers in test arrays
+    {
+      code: `
+ruleTester.run('test', rule, {
+  valid: [testCase],
+  invalid: [],
+});
+      `,
+      errors: [
+        {
+          column: 11,
+          line: 3,
+          messageId: 'noDynamicTests',
+        },
+      ],
+    },
+    // Template literals in test arrays
+    {
+      code: `
+ruleTester.run('test', rule, {
+  valid: [\`\${getTest()}\`],
+  invalid: [],
+});
+      `,
+      errors: [
+        {
+          column: 11,
+          line: 3,
+          messageId: 'noDynamicTests',
+        },
+      ],
+    },
+    // Binary expressions in test arrays
+    {
+      code: `
+ruleTester.run('test', rule, {
+  valid: ['test' + getSuffix()],
+  invalid: [],
+});
+      `,
+      errors: [
+        {
+          column: 11,
+          line: 3,
+          messageId: 'noDynamicTests',
+        },
+      ],
+    },
+    // Conditional expressions in test arrays
+    {
+      code: `
+ruleTester.run('test', rule, {
+  valid: [shouldTest ? 'test1' : 'test2'],
+  invalid: [],
+});
+      `,
+      errors: [
+        {
+          column: 11,
+          line: 3,
+          messageId: 'noDynamicTests',
+        },
+      ],
+    },
+    // Member expressions in test arrays
+    {
+      code: `
+ruleTester.run('test', rule, {
+  valid: [testConfig.cases],
+  invalid: [],
+});
+      `,
+      errors: [
+        {
+          column: 11,
+          line: 3,
+          messageId: 'noDynamicTests',
+        },
+      ],
+    },
+    // Object spread
+    {
+      code: `
+ruleTester.run('test', rule, {
+  valid: [{ ...testConfig }],
+  invalid: [],
+});
+      `,
+      errors: [
+        {
+          column: 11,
+          line: 3,
+          messageId: 'noDynamicTests',
+        },
+      ],
+    },
+  ],
+  valid: [
```

Comment:
[Testing] TaggedTemplateExpressions like `noFormat` should have test coverage to make sure that's allowed:

```ts
    {
      code: `
ruleTester.run('test', rule, {
  valid: [noFormat\`const x = 1;\`],
  invalid: [],
});
      `,
    },
```